### PR TITLE
Feature/build and publish provider

### PR DESCRIPTION
**Pull request by marns93**
I've added the new CI workflow and used your reusable workflow.
The main workflow runs without any problems and the artifacts are stored properly on GitHub, see https://github.com/pulumiverse/pulumi-heroku/actions/runs/2833362034

I would like to publish the provider to PyPI like https://pypi.org/search/?q=pulumiverse. The release workflow is added, is it possible to add the correct credentials to this repository?

I will test the publishing process afterwards.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+  PYPI_USERNAME: "__token__"
+  PYPI_REPOSITORY_URL: ""
+  PUBLISH_PYPI: true
+jobs:
+  publish_binary:
+    strategy:
+      fail-fast: true
+      matrix:
+        goversion:
+          - 1.17.x
```
Comment (ringods) [inline .github/workflows/release.yml:37]: Please bump to Go 1.18.x. All of Pulumi is already on that version.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+          - 3.1.301
+        goversion:
+          - 1.17.x
+        language:
+          - nodejs
+          - python
+          - dotnet
+          - go
+        nodeversion:
+          - 14.x
```
Comment (ringods) [inline .github/workflows/release.yml:83]: Please bump to NodeJS 16. v14 is way beyond end-of-life.

**Review by ringods — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml (hunk 1 above)
Comment (ringods) [reply .github/workflows/release.yml:37]: See this PR as a similar upgrade: https://github.com/pulumiverse/pulumi-time/pull/4

After changing the Go version in `go.mod` files, run `go mod tidy` for the checksum files to be regenerated.

Comment (marns93) [reply .github/workflows/release.yml]: Thanks, solved in https://github.com/pulumiverse/pulumi-heroku/pull/4/commits/c990017b4b449e659eefcbaa877afdd4a6fd2283

File: .github/workflows/release.yml (hunk 2 above)
Comment (marns93) [reply .github/workflows/release.yml]: Solved in https://github.com/pulumiverse/pulumi-heroku/pull/4/commits/c990017b4b449e659eefcbaa877afdd4a6fd2283

**Review by ringods — APPROVED** [review]

**Review by ringods — CHANGES_REQUESTED** [review]
Can you also add the Go 1.18 workspace files in the root of the repo?
`go.work` and `go.work.sum`
See here for an example: https://github.com/pulumiverse/pulumi-time/blob/main/go.work

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+  # NPM REGISTRY THEN ENSURE THE NPM_REGISTRY_URL IS CHANGED
+  NUGET_PUBLISH_KEY: ${{ secrets.NUGET_PUBLISH_KEY }}
+  NUGET_FEED_URL: https://api.nuget.org/v3/index.json
+  PUBLISH_NUGET: true
+  # IF YOU NEED TO PUBLISH A PYPI PACKAGE THEN ENSURE AN PYPI_API_TOKEN
+  # SECRET IS SET AND PUBLISH_PYPI: TRUE. IF YOU WANT TO PUBLISH TO AN ALTERNATIVE
+  # PYPI REGISTRY THEN ENSURE THE PYPI_REPOSITORY_URL IS SET. IF YOU ARE USING AN API_TOKEN THEN
+  # YOU DO NOT NEED TO CHANGE THE PYPI_USERNAME (__token__) , IF YOU ARE USING PASSWORD AUTHENTICATION THEN YOU WILL
+  # NEED TO CHANGE TO USE THE CORRECT PASSWORD
+  PYPI_PASSWORD: ${{ secrets.PYPI_API_TOKEN }}
```
[the remark this answers is not in the document]
Comment (marns93) [reply .github/workflows/release.yml]: Thanks. I've updated the variables and added the files. 

Publish successful. :)
See: https://pypi.org/project/pulumiverse-heroku/

**Review by ringods — APPROVED** [review]

**Pull request merged**